### ISSUE TITLE
docker-compose up not working

### ISSUE THREAD

**Issue by ineiti** [c4dt/service-stainless#114]
When I clone the repo and type `docker-compose up`, it doesn't work:

```
% docker-compose up
[+] Running 0/5
 ⠇ conode-1 Pulling                                                                                         3.8s
 ⠇ conode-2 Pulling                                                                                         3.8s
 ⠇ conode-4 Pulling                                                                                         3.8s
 ⠇ conode-3 Pulling                                                                                         3.8s
 ⠿ demo Warning                                                                                             3.4s
[+] Building 0.9s (17/33)                                                                                        
 => [c4dt/service-stainless-demo:latest internal] load build definition from Dockerfile                     0.0s
 => => transferring dockerfile: 32B                                                                         0.0s
 => [c4dt/service-stainless-backend:latest internal] load build definition from Dockerfile                  0.0s
 => => transferring dockerfile: 32B                                                                         0.0s
 => [c4dt/service-stainless-demo:latest internal] load .dockerignore                                        0.0s
 => => transferring context: 34B                                                                            0.0s
 => [c4dt/service-stainless-backend:latest internal] load .dockerignore                                     0.0s
 => => transferring context: 2B                                                                             0.0s
 => [c4dt/service-stainless-demo:latest internal] load metadata for docker.io/joseluisq/static-web-server:  0.6s
 => [c4dt/service-stainless-demo:latest internal] load metadata for docker.io/library/node:16-alpine        0.7s
 => [c4dt/service-stainless-backend:latest internal] load metadata for docker.io/library/debian:bookworm-s  0.7s
 => [c4dt/service-stainless-backend:latest internal] load metadata for docker.io/library/golang:1           0.5s
 => CANCELED [c4dt/service-stainless-demo:latest runner 1/3] FROM docker.io/joseluisq/static-web-server:2@  0.1s
 => => resolve docker.io/joseluisq/static-web-server:2@sha256:46ff7a766b971ce3c91bedd08da5fa3ef6fe4d450623  0.0s
 => => sha256:46ff7a766b971ce3c91bedd08da5fa3ef6fe4d450623bb8fe9c9da88214a8738 3.91kB / 3.91kB              0.0s
 => => sha256:74d1c512f5b91786b84f026d22fe48261a08fe26640d7b4758d07e3bdcd044ac 671B / 671B                  0.0s
 => => sha256:e017b38232a5c5953ee00067bdf4b679f99ccb4c6fb213ef335d1c374d0ee3f1 3.29kB / 3.29kB              0.0s
 => CANCELED [c4dt/service-stainless-demo:latest builder 1/6] FROM docker.io/library/node:16-alpine@sha256  0.1s
 => => resolve docker.io/library/node:16-alpine@sha256:f1657204d3463bce763cefa5b25e48c28af6fe0cdb0f68b354f  0.0s
 => => sha256:0ac2f8cba4274f3aeeb8dd2cde7714fe746b917df68a79eb9b4d5de053c5aa53 1.16kB / 1.16kB              0.0s
 => => sha256:5b6b1b14e0516d9de0a964a1e3022e17d517539b3e37082478f7adcaf1e2af95 6.50kB / 6.50kB              0.0s
 => => sha256:f1657204d3463bce763cefa5b25e48c28af6fe0cdb0f68b354f0f8225ef61be7 1.43kB / 1.43kB              0.0s
 => [c4dt/service-stainless-demo:latest internal] load build context                                        0.0s
 => => transferring context: 2.67kB                                                                         0.0s
 => CANCELED [c4dt/service-stainless-backend:latest builder 1/7] FROM docker.io/library/golang:1@sha256:6d  0.1s
 => => resolve docker.io/library/golang:1@sha256:6dd5c5f8936d7d4487802fb10a77f31b1776740be0fc17ada1acb74ac  0.0s
 => => sha256:83b2150ac94d2d1f797a8be5bf9ba2e63a67e359ca3269f90f4285f409306212 7.33kB / 7.33kB              0.0s
 => => sha256:6dd5c5f8936d7d4487802fb10a77f31b1776740be0fc17ada1acb74ac958f7be 2.36kB / 2.36kB              0.0s
 => => sha256:f8b1448cc5c46a269364238c52fd58919e16978a9f7987e325e645c2edf23f2f 1.79kB / 1.79kB              0.0s
 => [c4dt/service-stainless-backend:latest internal] load build context                                     0.0s
 => => transferring context: 79.08kB                                                                        0.0s
 => CANCELED [c4dt/service-stainless-backend:latest runner 1/7] FROM docker.io/library/debian:bookworm-sli  0.1s
 => => resolve docker.io/library/debian:bookworm-slim@sha256:2fd76b2b011a8b171f9ee0540fb74e9be0dc55fafc8f7  0.0s
 => => sha256:055962183328d0da6ca0596ccddaeceb968bdb51a2e37a8f371fab8741a23641 1.48kB / 1.48kB              0.0s
 => => sha256:2fd76b2b011a8b171f9ee0540fb74e9be0dc55fafc8f7f2858a482c0e7776ebc 1.85kB / 1.85kB              0.0s
 => => sha256:e6398a7a1e9fb3b33a923b1e7a1f0576887939543a2f34d6b0bb02ae1a4ce732 529B / 529B                  0.0s
 => CACHED [c4dt/service-stainless-backend:latest runner 2/7] RUN apt update     && apt install --no-insta  0.0s
 => CACHED [c4dt/service-stainless-backend:latest runner 3/7] RUN npm install --global solc@0.5             0.0s
 => ERROR [c4dt/service-stainless-backend:latest runner 4/7] COPY build/cvc4 /usr/local/bin/                0.0s
 => ERROR [c4dt/service-stainless-backend:latest runner 5/7] COPY build/stainless.zip /tmp/                 0.0s
------
 > [c4dt/service-stainless-backend:latest runner 4/7] COPY build/cvc4 /usr/local/bin/:
------
------
 > [c4dt/service-stainless-backend:latest runner 5/7] COPY build/stainless.zip /tmp/:
------
failed to solve: failed to compute cache key: "/build/cvc4" not found: not found
```

**Comment by PascalinDe** [comment]
I have different issues:

* `docker-compose pull` is not working: `* Error response from daemon: pull access denied for c4dt/service-stainless-demo, repository does not exist or may require 'docker login': denied: requested access to the resource is denied`
* trying to build the images locally is not working either:

```
 > [demo builder 6/6] RUN npx ng build --prod:                                                                                              
#0 21.18                                                                                                                                    
#0 21.18 chunk {0} runtime.7d9d4678f9511567f211.js (runtime) 1.45 kB [entry] [rendered]                                                     
#0 21.18 chunk {1} main.5aed90432064e81aaa2f.js (main) 128 bytes [initial] [rendered]                                                       
#0 21.18 chunk {2} polyfills.42d2fdf5686c739360fe.js (polyfills) 130 bytes [initial] [rendered]                                             
#0 21.18 chunk {3} polyfills-es5.10c83e44eb45a67bdc5d.js (polyfills-es5) 83.2 kB [initial] [rendered]
#0 21.18 chunk {4} styles.55ad0ea3aa75c62d700f.css (styles) 61 kB [initial] [rendered]
#0 21.18 Date: 2023-05-22T14:18:34.669Z - Hash: 712824b6a8d0b3b1fca9 - Time: 14970ms
#0 21.18 
#0 21.18 ERROR in lib/stainless/stainless-rpc.ts:5:36 - error TS2307: Cannot find module 'src/lib/proto'.
#0 21.18 
#0 21.18 5 import { stainless as proto } from "src/lib/proto";
#0 21.18                                      ~~~~~~~~~~~~~~~
#0 21.18 app/stainless/stainless.component.ts:9:36 - error TS2307: Cannot find module 'src/lib/proto'.
#0 21.18 
#0 21.18 9 import { stainless as proto } from "src/lib/proto";
#0 21.18                                      ~~~~~~~~~~~~~~~
#0 21.18 
------
failed to solve: process "/bin/sh -c npx ng build --prod" did not complete successfully: exit code: 1
```

**Comment by ineiti** [comment]
Check if `docker compose up` with a `docker-file.yaml` having both `image: ghcr.io....` and `build: .` in it does first a `pull` or starts `build`ing.

**Comment by ineiti** [comment]
Closed by #120
